disperser/dataapi/metrics.go

```diff
@@ -101,35 +104,124 @@ type DynamoDBCollector struct {
 	blobMetadataStore *blobstore.BlobMetadataStore
 	blobStatusMetric  *prometheus.Desc
 	logger            common.Logger
+	metricsCache      map[disperser.BlobStatus]int
+	cacheMutex        sync.RWMutex
 }
 
 func NewDynamoDBCollector(blobMetadataStore *blobstore.BlobMetadataStore, logger common.Logger) *DynamoDBCollector {
-	return &DynamoDBCollector{
+	collector := &DynamoDBCollector{
 		blobMetadataStore: blobMetadataStore,
 		blobStatusMetric: prometheus.NewDesc("dynamodb_blob_metadata_status_count",
 			"Number of blobs with specific status in DynamoDB",
 			[]string{"status"},
 			nil,
 		),
-		logger: logger,
+		logger:       logger,
+		metricsCache: make(map[disperser.BlobStatus]int),
 	}
+
+	if blobMetadataStore != nil {
+		go collector.periodicFetch()
+	} else {
+		logger.Warn("BlobMetadataStore is nil, metrics will not be collected")
+	}
+
+	return collector
 }
 
-func (collector *DynamoDBCollector) Describe(ch chan<- *prometheus.Desc) {
-	ch <- collector.blobStatusMetric
+func (collector *DynamoDBCollector) periodicFetch() {
+	// We don't want to fetch finalized status because the amount is large
+	statuses := []disperser.BlobStatus{disperser.Processing, disperser.Confirmed, disperser.InsufficientSignatures, disperser.Failed}
```

Comment:
Can they just be tracked with prometheus metrics?

Reply:
Whenever the pod restarts the Prometheus metrics will reset to 0, so we need to periodically fetch the total amount.